richtextfx/src/main/java/org/fxmisc/richtext/model/UndoActions.java

```diff
@@ -21,8 +20,8 @@
     default void redo() { getUndoManager().redo(); }
 
     default boolean isUndoAvailable() { return getUndoManager().isUndoAvailable(); }
-    default ObservableBooleanValue undoAvailableProperty() { return getUndoManager().undoAvailableProperty(); }
+    default Val<Boolean> undoAvailableProperty() { return getUndoManager().undoAvailableProperty(); }
 
     default boolean isRedoAvailable() { return getUndoManager().isRedoAvailable(); }
-    default ObservableBooleanValue redoAvailableProperty() { return getUndoManager().redoAvailableProperty(); }
+    default Val<Boolean> redoAvailableProperty() { return getUndoManager().redoAvailableProperty(); }
```

Comment:
I think I misled myself by using `C` as the type parameter instead of `?`. Hmm... Should the class-to-interface conversion for `TextChange` still be accepted though? The refactoring isn't necessary, but it might be forward-thinking.